OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -201,108 +211,80 @@ void ApplyOrders(World world, MouseInput mi)
 			}
 		}
 
-		public override string GetCursor(int2 screenPos)
-		{
-			return Sync.CheckSyncUnchanged(World, () =>
-			{
-				// Always show an arrow while selecting
-				if (SelectionBox != null)
-					return null;
-
-				var cell = worldRenderer.Viewport.ViewToWorld(screenPos);
-
-				var mi = new MouseInput
-				{
-					Location = screenPos,
-					Button = Game.Settings.Game.MouseButtonPreference.Action,
-					Modifiers = Game.GetModifierKeys()
-				};
-
-				return World.OrderGenerator.GetCursor(World, cell, mi);
-			});
-		}
-
 		public override bool HandleKeyPress(KeyInput e)
 		{
-			var player = World.RenderPlayer ?? World.LocalPlayer;
-
 			if (e.Event == KeyInputEvent.Down)
 			{
 				var key = Hotkey.FromKeyInput(e);
-
-				if (key == Game.Settings.Keys.PauseKey && World.LocalPlayer != null) // Disable pausing for spectators
-					World.SetPauseState(!World.Paused);
-				else if (key == Game.Settings.Keys.SelectAllUnitsKey && !World.IsGameOver)
-				{
-					// Select actors on the screen which belong to the current player
-					var ownUnitsOnScreen = SelectActorsOnScreen(World, worldRenderer, null, player).SubsetWithHighestSelectionPriority();
-					World.Selection.Combine(World, ownUnitsOnScreen, false, false);
-				}
-				else if (key == Game.Settings.Keys.SelectUnitsByTypeKey && !World.IsGameOver)
+				var hotkeyMap = new Dictionary<Hotkey, Func<bool>>
```

Comment:
C# won't let me
"Error CS0151 A switch expression or case label must be a bool, char, string, integral, enum, or corresponding nullable type"